### ISSUE TITLE
How to trains a 8 scales network?

### ISSUE THREAD

**Issue by Dogdog00** [idealo/image-super-resolution#127]
Hi:

The pictures I generated using my own data set are not very good. I want to know how to get a best 8 scales weights.

Thanks

**Comment by cfrancesco** [comment]
This question is a little too vague, please be more specific on what issues you're encountering.

**Comment by Dogdog00** [comment]
My training set is 1080 resolution pictures, and they are first downsampled and then partially blurred to generate 128 resolution pictures. Then use 128 resolution and 1080 resolution as training set for training. The network architecture used is RRDN, but for training it, the psrn value is always 20 to 21 particularly small, and the chromaticity of the generated picture is distorted.![image](https://user-images.githubusercontent.com/64968282/85029134-8874c000-b1ae-11ea-9353-7f3ef5027cab.jpeg)

**Comment by cfrancesco** [comment]
so it seems you're using only the gan loss. I would recommend to first train with the psnr loss and see the results afterafter 30-40 epochs. If that is going in the right direction you could either continue (with the open souirce dataset I trained until 70-90 epochs with PSNR) or you introduce the gan loss. If I remember correctly you just need to change the loss weights to do this, or don't pass a discriminator network. It's probably safest to simply set all losses to zero other than generator loss and than (maybe gradually) change this. 

You could also first try without blurrying the pictures.
Best of luck!

**Comment by Dogdog00** [comment]
Thank you very much for your guidance, I hope to communicate a lot in the future and make progress together.

**Comment by Dogdog00** [comment]
Hi, What do these parameters stand for, 

loss_weights =
{
  'generator': 0.8,
  'feature_extractor':0.0833 ,
  'discriminator': 0.01
} 

and how do you determine their values ​​during training? How were they determined during your training? Also, can you share your specific training details? For example, which one to train first, then how to train, thank you very much!

**Comment by cfrancesco** [comment]
Hi,
generator is the loss weights for the plain network, having this to 1 and the others to 0 will only do pixelwise MAE with the original image.
feature_extractor is the weight to the MAE loss using VGG extracted features from the upscaled pictres and VGG extracted features from the original image.
Discriminator is the GANs loss.
First use only generator, the use a combination. You have to experiment. I would always use first generator (and maybe some feature) and then 0 generator with feature and GANs

**Comment by Dogdog00** [comment]
Thank you very much! Under your guidance, I feel that my training is about
to succeed!

Francesco <notifications@github.com>于2020年6月22日 周一19:05写道：

**Comment by Dogdog00** [comment]
When only the mae loss training is performed, to what extent can the
comprehensive loss training be performed? The picture generated by my mae
loss is a little blurry, not very clear. Can the mae loss only achieve the
desired result?

Xun Wsng <gougou.xw@gmail.com>于2020年6月22日 周一19:07写道：

**Comment by cfrancesco** [comment]
Hi, no, only using mae you will get less defined pictures. You can try adding only the feature loss, if training with GANs is too difficult
